Library/Homebrew/dev-cmd/audit.rb

```diff
@@ -236,8 +236,29 @@ def self.check_http_content(url, name, user_agents: [:default])
       details[:content_length] == secure_details[:content_length]
     file_match = details[:file_hash] == secure_details[:file_hash]
 
-    return if !etag_match && !content_length_match && !file_match
-    "The URL #{url} could use HTTPS rather than HTTP"
+    if etag_match || content_length_match || file_match
+      return "The URL #{url} could use HTTPS rather than HTTP"
+    end
+
+    return if type != "homepage"
+
+    details[:file] = details[:file].gsub(%r{https?:\\?\/\\?\/}, "/")
+    secure_details[:file] = secure_details[:file].gsub(%r{https?:\\?\/\\?\/}, "/")
+
+    # Same content after normalization
+    if details[:file] == secure_details[:file]
+      return "The URL #{url} could use HTTPS rather than HTTP"
+    end
+
+    # Same size, different content after normalization
+    # (typical causes: Generated ID, Timestamp, Unix time)
+    if details[:file].length == secure_details[:file].length
```

Comment:
I think both this check and the one below should be only for `--strict` (maybe `--new-formula`?) because they (presumably) will produce some false positives. @ilovezfs, thoughts?

Reply:
If there is precedent for an audit _warning_, it could be used here. Per past experience a length match at this point is an pretty-sure bet for upgradability.